lib/smart_routing/user/object.rb

```diff
@@ -0,0 +1,36 @@
+module SmartRouting
```

Comment:
As I see, u have 4 very similar classes of `SmartRouting::User`. Maybe it's better to add `Base` class and inherit `Object`, `ObjectType`, `Rule` and `Set`?

Reply:
> As I see, u have 4 very similar classes of `SmartRouting::User`. Maybe it's better to add `Base` class and inherit `Object`, `ObjectType`, `Rule` and `Set`?

I don't want add `Base` class only for `initialize` method :)